File: basic-proxy-impl/src/test/java/com/test/cases/interfaces/RunnableTest.java
package com.test.cases.interfaces;

import org.junit.Test;

public class RunnableTest extends InterfaceTestBase<Runnable> {

    private static final Class<Runnable> specifiedInterface = Runnable.class;

    @Test
    public void test() throws Exception {
        Runnable standardProxy = buildStandardProxy(specifiedInterface);
        standardProxy.run();

        Runnable customProxy = buildCustomProxy(specifiedInterface);
        customProxy.run();

        verify(specifiedInterface.getMethod("run"));
    }
}

File: basic-proxy-kotlin-impl/src/test/kotlin/com/test/cases/interfaces/RunnableTest.kt
package com.test.cases.interfaces

import org.junit.Test

class RunnableTest : InterfaceTestBase<Runnable>() {

    private val specifiedInterface = Runnable::class.java

    @Test
    fun test() {
        val standardProxy = buildStandardProxy(specifiedInterface)
        standardProxy.run()

        val customProxy = buildCustomProxy(specifiedInterface)
        customProxy.run()

        verify(specifiedInterface.getMethod("run"))
        verifyFields(standardProxy, customProxy)
    }
}
